unittests/test_deprecation.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Insight Software Consortium.
+# Distributed under the Boost Software License, Version 1.0.
+# See http://www.boost.org/LICENSE_1_0.txt
+
+import unittest
+import pprint
+
+from . import parser_test_case
+
+from pygccxml import parser
+from pygccxml import declarations
+
+
+class Test(parser_test_case.parser_test_case_t):
+    global_ns = None
+
+    def __init__(self, *args):
+        parser_test_case.parser_test_case_t.__init__(self, *args)
+        self.header = "test_deprecation.hpp"
+        self.global_ns = None
+        self.config.castxml_epic_version = 1
+
+    def _check_text_content(self, desired_text, deprecation_string):
+        if deprecation_string:
+            self.assertEqual(desired_text, deprecation_string)
+        else:
+            pprint.pprint("No text in deprecation string")
```

Comment:
Why `pprint` and not normal `print`? These also aren't showing up in the CI, but that's expected since we're using `castxml_epic_version = 1`

Reply:
Sorry @tao558, I assumed something and that was not helpful.  I think its the multiple layers of unittest that causes the print output to be missing when we run ``test_all``.  They show up when the test is run directly. I'm not sure how to pipe them upwards.  I'll revert the ``pprint`` and call this ready again.

Reply:
No problem at all. So these print statements are supposed to be in the output, which means none of the deprecation string values are being checked. Is this because of the version of castxml we're using?

Reply:
No.  The printed text should only show up when a version of CastXML is used that isn't the most recent release (ie. before we changed the CI download hash) .   It's to prevent the test from throwing an error when the content isn't present even though it really isn't there.  

I was unable to find a good way to prevent the test from being run if the CastXML version wasn't sufficiently new.